int32 NAO
---
int32 arm
